src/schema/__tests__/ecommerce/create_offer_order_with_artwork_mutation.test.ts

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable promise/always-return */
+import { runQuery } from "test/utils"
+import { mockxchange } from "test/fixtures/exchange/mockxchange"
+import sampleOrder from "test/fixtures/results/sample_order"
+import exchangeOrderJSON from "test/fixtures/exchange/order.json"
+import gql from "lib/gql"
+import { OrderBuyerFields } from "./order_fields"
+
+let rootValue
+
+describe("Create Offer Order Mutation", () => {
+  beforeEach(() => {
+    const resolvers = {
+      Mutation: {
+        createOfferOrderWithArtwork: () => ({
+          orderOrError: exchangeOrderJSON,
+        }),
+      },
+    }
+
+    rootValue = mockxchange(resolvers)
+  })
+
+  it("creates offer order and returns it", () => {
+    const mutation = gql`
+      mutation {
+        ecommerceCreateOfferOrderWithArtworkMutation(
+          input: { artworkId: "111", editionSetId: "232", quantity: 1 }
+        ) {
+          orderOrError {
+            ... on Order {
+              ${OrderBuyerFields}
+            }
+            ... on EcommerceError {
+              type
+              errorCode: code
+              data
+            }
+          }
+        }
+      }
+    `
+
+    return runQuery(mutation, rootValue).then(data => {
+      expect(
+        data!.ecommerceCreateOfferOrderWithArtworkMutation.orderOrError
+      ).toEqual(sampleOrder(true, false))
```

Comment:
I find the tests that hide the output in a fixture to be a little magical for my taste.

But the real comment-- in this case, it would be good to assert both the "success" case and the "failure" cases. At this layer we could test for known errors from exchange and errors _reaching_ exchange at all (or a generic 500) since they have different behavior.